server/game/cards/02_SHD/upgrades/MandalorianArmor.ts

```diff
@@ -0,0 +1,28 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+import { Trait } from '../../../core/Constants';
+import { Card } from '../../../core/card/Card';
+
+export default class MandalorianArmor extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '3514010297',
+            internalName: 'mandalorian-armor'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setAttachCondition((card: Card) => !card.hasSomeTrait(Trait.Vehicle));
+
+        this.addWhenPlayedAbility({
+            title: 'If attached unit is a Mandalorian, give a Shield token to it.',
+            immediateEffect: AbilityHelper.immediateEffects.conditional((context) => ({ target: context.source.parentCard,
```

Comment:
I am actually not 100% sure that using `target` in `conditional()` works correctly in all cases, do you mind moving this inside of `giveShield()`?